MembraneRTC/Sources/MembraneRTC/MembraneRTC.swift

```diff
@@ -74,15 +73,15 @@ public class MembraneRTC: MulticastDelegate<MembraneRTCDelegate>, ObservableObje
     // mapping from remote track's id to its context
     private var trackContexts: [String: TrackContext] = [:]
 
-    // mapping from transceiver's mid to its remote track id
-    private var midToTrackId: [String: String] = [:]
-
     // receiver used for iOS screen broadcast
     private var broadcastScreenshareReceiver: ScreenBroadcastNotificationReceiver?
 
-    private static let mediaConstraints = RTCMediaConstraints(
-        mandatoryConstraints: nil,
-        optionalConstraints: ["DtlsSrtpKeyAgreement": kRTCMediaConstraintsValueTrue])
+    private var encoder: Encoder
+
+    private lazy var peerConnectionManager: PeerConnectionManager = {
+        return PeerConnectionManager(
+            config: self.config, peerConnectionFactory: self.peerConnectionFactoryWrapper, peerConnectionListener: self)
+    }()
```

Comment:
No :D that's a problem with swift: you can't pass `self` as an argument to initialize `PeerConnectionManager` because you'll get an error:
![Screenshot 2022-12-06 at 01 43 40](https://user-images.githubusercontent.com/6280457/205778102-676500c7-75e4-4dcd-abff-c98b4296baf1.png)
And no, you can't move it after `super.init()`:
![Screenshot 2022-12-06 at 01 43 55](https://user-images.githubusercontent.com/6280457/205778154-fb700a28-2c77-4e8f-b0bf-9498493195b9.png)
And it's only because of that I used lazy, we don't really need to wait on anything and in any other language I know you can do this without such workarounds. Maybe there is some more elegant way, I can search a little more.